### ISSUE TITLE
Complexity of `add_stat`

### ISSUE THREAD

**Issue by jmid** [c-cube/qcheck#41]
This can be considered a separate issue to #40 hence the separate report (but feel free to merge).

If I try to inspect the (unsigned) `int` generator (by wrapping it with `abs`) it takes a very long time 
(I still haven't waited it out) to compute statistics of a sample of (the default) 100 integers:
```
# let t = Test.make (add_stat ("dist",abs) int) (fun _ -> true);;
val t : QCheck.Test.t = QCheck.Test.Test <abstr>
# QCheck_runner.run_tests ~verbose:true [t];;                    
generated error  fail pass / total  time(s) name
[✓]  100    0     0    100 /  100      0.0  anon_test_12                                            

(top-level hangs)
```
This behaviour indicates that the 100 `true` tests completed immediately, while computing/drawing statistics take substantial time. I suspect if one of these routines algorithmically "count up" from 0, then it will take a long time to reach the entries produced by `int` (wrapped in `abs`) - but that is just a guess without having studied the code thoroughly.

**Comment by c-cube** [comment]
Indeed, there is a loop on every index. It should instead become a loop on the table `value -> num of occurrences` (which can become quite sparse if values range over a wide domain). This requires to pre-allocate the buckets though, so it will take a bit of time, I can't do it right no.

**Comment by c-cube** [comment]
Fixed by #45, as tests can show. Thanks @jmid!
